src/momento/momento_signer.py

```diff
@@ -0,0 +1,140 @@
+import json
+from enum import Enum
+from typing import Optional, Dict
+from jwt.api_jwk import PyJWK
+import jwt
+
+from .errors import InvalidArgumentError
+
+
+class CacheOperation(Enum):
+    GET = 1
+    SET = 2
+
+
+class SigningRequest:
+    def __init__(
+        self,
+        cache_name: str,
+        cache_key: str,
+        cache_operation: CacheOperation,
+        expiry_epoch_seconds: int,
+        ttl_seconds: Optional[int] = None,
+    ):
+        """Initializes a new SigningRequest.
+
+        Args:
+            cache_name: The name of the cache.
+            cache_key: The key of the object.
+            cache_operation: The operation performed on the item in the cache.
+            expiry_epoch_seconds: The timestamp that the pre-signed URL is valid until.
+            ttl_seconds: Time to Live for the item in Cache. This is an optional property that will only be used for CacheOperation.SET.
+        """
+        self._cache_name = cache_name
+        self._cache_key = cache_key
+        self._cache_operation = cache_operation
+        self._expiry_epoch_seconds = expiry_epoch_seconds
+        self._ttl_seconds = ttl_seconds
+
+    def expiry_epoch_seconds(self) -> int:
+        return self._expiry_epoch_seconds
+
+    def cache_name(self) -> str:
+        return self._cache_name
+
+    def cache_key(self) -> str:
+        return self._cache_key
+
+    def cache_operation(self) -> CacheOperation:
+        return self._cache_operation
+
+    def ttl_seconds(self) -> Optional[int]:
+        return self._ttl_seconds
+
+
+class MomentoSigner:
+    def __init__(self, jwk_json_string: str):
+        """Initializes MomentoSigner with the specified private key.
+
+        Args:
+            jwk_json_string: the JSON string of the JWK key. This can be obtained from create_signing_key response.
+
+        Raises:
+            InvalidArgumentError: If the supplied private key is not valid.
+        """
+        try:
+            jwk_json: Dict[str, str] = json.loads(jwk_json_string)
+        except json.decoder.JSONDecodeError:
+            raise InvalidArgumentError(f"Invalid JWK Json String: {jwk_json_string}")
+
+        if "kid" not in jwk_json:
+            raise InvalidArgumentError(f"JWK missing kid attribute: {jwk_json_string}")
+
+        if "alg" not in jwk_json:
+            raise InvalidArgumentError(f"JWK missing alg attribute: {jwk_json_string}")
```

Comment:
This is not illegal. It could have crv and/or kty.

Reply:
Hmm this is a tricky one. We know the JWKs we return from our service do have 'alg' defined though.

pyjwt library that we are using does have the logic to handle when `alg` is missing but it's not exposed to us unfortunately: https://github.com/jpadilla/pyjwt/blob/master/jwt/api_jwk.py#L19

I might need to copy-n-paste that logic into our sdk here.